### PR TITLE
docs(plugins): update option type

### PR DESCRIPTION
**Pull request by chenxsan**
I think `(string | RegExp)[]` is accurate than `Array`.

### CODE REVIEW

File: src/content/plugins/banner-plugin.md [hunk 1]
```diff
@@ -28,9 +29,9 @@ new webpack.BannerPlugin(options);
   banner: string | function, // the banner as string or function, it will be wrapped in a comment
   raw: boolean, // if true, banner will not be wrapped in a comment
   entryOnly: boolean, // if true, the banner will only be added to the entry chunks
-  test: string | RegExp | Array,
-  include: string | RegExp | Array,
-  exclude: string | RegExp | Array,
+  test: string | RegExp | (string | RegExp)[],
+  include: string | RegExp | (string | RegExp)[],
+  exclude: string | RegExp | (string | RegExp)[],
```
Comment (snitin315) [inline src/content/plugins/banner-plugin.md:34]:
```suggestion
  test: string | RegExp | [string, RegExp],
  include: string | RegExp | [string, RegExp],
  exclude: string | RegExp | [srting, RegExp],
```

Comment (chenxsan) [reply src/content/plugins/banner-plugin.md:34]: Thanks for the link. But here's my reason, although both are valid in typescript/flow, `[string, RegExp]` means totally different thing from `(string | RegExp)[]`. Now that webpack is going to include [types for typescript](https://github.com/webpack/webpack/pull/10599), I think it's better to follow typescript's conventions.

Maybe we shall update the writers' guide?

**Comment by jeffin143** [comment]
Should we merge #3693 , so that both could be reviewed in one go

I could fetch your branch make changes there , or you could fetch mine 
Feel free to let me know :)

File: src/content/plugins/banner-plugin.md (hunk 1 above)
Comment (montogeek) [reply src/content/plugins/banner-plugin.md:34]: Interesting, would be great to match source types agaisnt the docs, automatically.

**Comment by chenxsan** [comment]
@jeffin143 Thanks for suggestion, but these two seem to be totally different things although they target the same file, maybe better keep them separated.

**Comment by chenxsan** [comment]
Found some discussions about how to present types in doc here https://github.com/webpack/webpack.js.org/pull/3553/files/7c05c8994023d71fde997dee218774bc65952665#r380507188, and [from that example](https://webpack.js.org/configuration/entry-context/#entry), I would say the current guides https://webpack.js.org/contribute/writers-guide/#configuration-defaults-and-types is kind of broken when dealing with complex cases:

image

I don't think anyone can figure out this easily.

Any here's the typescript version formatted with prettier:

```typescript
type Entry =
  | string
  | [string]
  | {
      key:
        | string
        | [string]
        | {
            import: string | [string]
            dependOn: string | [string]
            filename: string
          }
    }
  | (() =>
      | string
      | [string]
      | {
          key:
            | string
            | [string]
            | {
                import: string | [string]
                dependOn: string | [string]
                filename: string
              }
        })
```
It's much longer, but it's clean, and much easy to read.

File: src/content/plugins/banner-plugin.md (hunk 1 above)
Comment (EugeneHlushko) [reply src/content/plugins/banner-plugin.md:34]: Hi @chenxsan 
Its great that you suggest improvements, but currently to express an array of items limited to strings and RegExps we use `[string, RegExp]` in the documentation. We could continue the discussion of changing the types across the docs to be in TypeScript in the existing issue https://github.com/webpack/webpack.js.org/issues/3333

**Comment by EugeneHlushko** [comment]
Thank you!

**Pull request merged**